app/substitution_helper.rb

```diff
@@ -0,0 +1,42 @@
+module SubstitutionHelper
+  class << self
+    def clear_draft!(draft_content_item)
+      if clear!(draft_content_item, DraftContentItem)
+        clear_foreign_key!(draft_content_item)
+      end
+    end
+
+    def clear_live!(draft_content_item)
+      clear!(draft_content_item, LiveContentItem)
+    end
+
+  private
+    def clear!(draft_content_item, klass)
+      existing_content_item = klass.find_by(base_path: draft_content_item.base_path)
+      return unless existing_content_item
+
+      mismatch = existing_content_item.content_id != draft_content_item.content_id
+      allowed_to_substitute = substitute?(draft_content_item) || substitute?(existing_content_item)
```

Comment:
Do you want to wrap these in parens?  I know it's not strictly required but reads a little easier.